File: backend/EcommerceApi/Models/producto_model.cs
namespace EcommerceApi.Models;

public class Producto
{
    public int Id { get; set; }
    public string Nombre { get; set; } = string.Empty;
    public string Descripcion { get; set; } = string.Empty;
    public decimal Precio { get; set; }
    public int Stock { get; set; }
    public string ImagenUrl { get; set; } = string.Empty;
    public bool Activo { get; set; } = true;
    public DateTime FechaCreacion { get; set; } = DateTime.UtcNow;
    
    // Relación con Categoría
    public int CategoriaId { get; set; }
    public Categoria? Categoria { get; set; }
    
    // Relaciones
    public ICollection<CarritoItem> CarritoItems { get; set; } = new List<CarritoItem>();
    public ICollection<PedidoItem> PedidoItems { get; set; } = new List<PedidoItem>();
}

File: frontend/ecommerce-app/src/app/shared/models/producto.model.ts
export interface Producto {
  id: number;
  nombre: string;
  descripcion: string;
  precio: number;
  stock: number;
  imagenUrl: string;
  imagenUrl2?: string;
  imagenUrl3?: string;
  imagenes?: string[];
  activo: boolean;
  categoriaId: number;
  categoria?: Categoria;
}

export interface Categoria {
  id: number;
  nombre: string;
  descripcion: string;
}

export interface ProductoCreateDto {
  nombre: string;
  descripcion: string;
  precio: number;
  stock: number;
  imagenUrl: string;
  imagenUrl2?: string;
  imagenUrl3?: string;
  categoriaId: number;
}
